### DISCUSSION TITLE
Unicode Problem in a File Path

### DISCUSSION THREAD

**Discussion by ya158** [iBotPeaches/Apktool#3704] [Q&A]
I'm trying to unpack a file with non-unicode characters in the path:
`java.exe -jar -Duser.language=en -Dfile.encoding=UTF8  "d:\Poco\Темы\Шаблоны\Конструктор шаблонов\Конструктор\APK_unpacked\Bluetooth" "d:\Poco\Темы\Шаблоны\Конструктор шаблонов\Конструктор\APK\Bluetooth.apk"`
and I get an error:
Input file (D:\Poco\в??Т?В?Р?\В?В Р?В?В?В?Р?\Р?В?В?Р?Р?Р°Р?Р?Р?В?Р° РёВ Р?В?В?В?В?С?\Р?В?В?Р?Р?Р°Р?Р?Р?В?Р°\APK\Bluetooth.apk) was not found or was not readable.
How to unpack a file with non-unicode characters in the path?

**Comment by iBotPeaches** [comment]
Generally Apktool supports non-unicode characters, but downstream tools like aapt2 struggle with them. If you've found thats an issue in my implementation of the calling of aapt/aapt2 - then I'll accept patches.

However, I believe this is out of my control. So my workaround for you is to not have non-unicode characters int he path.

**Comment by EvenAngelsNeed** [comment]
I found this question after searching on "aapt.exe issue file path utf-8" since I noticed aapt has a problem with certain file path characters.

The solution I found in my case was to pass the short name (as opposed to windows long name) circumventing any utf-8 characters.

As IBotPeaches has suggested this issue can only be resolved directly by the aapt developers.

Mine is a workaround (for most apps that have this issue) but I hope this helps someone.
